### ISSUE TITLE
DrawOptions: bondLineWidth behaving differently since 2020 versions

### ISSUE THREAD

**Issue by kienerj** [rdkit/rdkit#3305]
**Configuration:**

- RDKit Version: 2020.03.1, 2020.03.3
- Operating system: Windows 10, Ubuntu 16.04
- Python version (if relevant): 3.7
- Are you using conda? Yes
- If you are using conda, which channel did you install the rdkit from? rdkit

**Description:**

The behavior of bondLineWidth has changed with versions from 2020 both on Windows and Linux (Ubuntu). Most notably setting bondLineWidth to 1 seems to not have an effect (looks like default which previously was 2)

```
from rdkit import Chem
from IPython.display import SVG

mol = Chem.MolFromSmiles('c1cc(CCC(O)CC=O)ccc1')
mol = Chem.Draw.rdMolDraw2D.PrepareMolForDrawing(mol, kekulize=True, addChiralHs=True)
drawer = Chem.Draw.rdMolDraw2D.MolDraw2DSVG(300, 150)
drawer.SetFontSize(0.75)       
opts = drawer.drawOptions()        
opts.bgColor = None
opts.clearBackground = False
opts.bondLineWidth = 1
#opts.fixedBondLength = 20

drawer.DrawMolecule(mol)
drawer.FinishDrawing()
svg = drawer.GetDrawingText() 
SVG(svg)
```

I noticed this when I wanted to play with the new "fixed size option" eg `fixedBondLength` which I have to say is extremely helpful to not have hugely drawn molecules (i the differ a lot in dimensions). Note that it's commented out but more on that later.

The behavior seems to be linked to this new setting. In above example bondLineWidth is same for values 0-5. if you set it to 6, then the bonds get thicker but they always have at least the thickness of this "lowest bin".

Interestingly if you uncomment `fixedBondLength` and set it to 20, the bonds get thicker. And now they stay at that thickness between 0-7 and only at 8 do they actually get thicker. Depending on `fixedBondLength` this threshold value changes. 

This is really confusing as bondLineWidth in my opinion should be a hard setting like before (or it appeared to be at least).
Bonds should not get thicker if I set `fixedBondLength` to a lower value. The bond thickness is a choice of style. (In my case the wish of the users so that structures are more "chemdraw" like)
Also the minimal thickness for bonds is now much thicker than before, closer to the previous versions default of 2 and can't be set any lower.

Attached is a direct comparisons from structure drawings in an application. right is rdkit 2019 and left is rdkit 2020.3.3. Only thing different is the update of rdkit (or more specifically `conda update rdkit -c rdkit`) which updated much more than rdkit but I do really think this is an rdkit issue and not 3rd party.

![rdkit bondLineWidth](https://user-images.githubusercontent.com/1567309/88511537-462c7300-cfe5-11ea-9f79-757d7496dc7e.png)

**Comment by DavidACosgrove** [comment]
There has indeed been a change in behaviour - the bond width now scales with the rest of the drawing.  I did this because I think it looks a bit daft to have a very large drawing with very thin bonds.  That's just my taste, though.  The minimum bond width is set at 2 of some unit that is probably related to pixels.  If you set opts.bondLineWidth to 0, it will always be this minimum value, because you obviously can't scale zero, so it will always default to the minimum of 2.  I think the behaviour you are seeing with the widths not changing for different values of bondLineWidth is because it is being set to the minimum value up until the point where the scaled width goes beyond that.  I guess you could make a case for an option to change this minimum value, and also for an option not to scale the line width from the value set.

**Comment by kienerj** [comment]
Hi David,

while personally I actually also prefer the thicker bonds, it's subjective and others seem to prefer the more ChemDraw like thin bonds (well depends on the ChemDraw style really but the style they prefer). My below text may seem "demanding". I'm aware this is an open-source project with the according limited resources. I'm mostly pointing out how according to my opinion it should ideally be. But it's just one opinion.

From a user and usability point of view I'm not very happy with the change. It breaks backwards compatibility while not being listed in the release as such. While it's ok that by default the bond width is scaled automatically, if the user (developer) sets it explicitly, it should be deterministic, eg be exactly that value regardless if the end results looks terrible or regardless of other options. Right now it's simply confusing as a certain range doesn't have any effect at all and that range is influence yet by other options. 

I'm also wondering what 0 should do. Maybe this should throw an exception as bond width of 1 (pixel?) should be the minimum? Or simply not draw any bonds? Anything below 0 should in any case throw an exception as it's nonsensical. Currently setting a negative number seems to simply have no effect. This is borderline exception swallowing, not very nice in my opinion. Invalid input should always throw an exception instead of reverting to some default.

My proposal:
- Setting `bondLineWidth` explicitly sets it to that exact fixed value - no other option needed (it's als compatible with older releases)
  - Edge case 0: Draw no bonds (maybe someone with a weird use case will need this?, also fine with exception)
  - negative number: throw an exception  (ValueError in python)

I suspect this thickness scaling is probably a requirement from MedChemica? So if this is needed and according to your suggested additional boolean option of scaling bond width, my proposal would still be as above but if `bondLineWidth` is set explicitly, the boolean scaling option, if not set explicitly, is false. Hence you get my suggested behavior. If you want different bond width plus scaling, you set `bondLineWidth` to desired value and `scaleBondWidth` (or how you want to call it) to true.
Albeit a bit confusing, this will again be backwards compatible to versions prior 2020.

**Comment by DavidACosgrove** [comment]
Hi @kienerj,
Many thanks for your thoughtful suggestions.  I think what you propose makes sense.  When I dug through the code for specifying the line width it was a bit convoluted and certainly would stand a some tidying up.  When I next get a wet weekend when I don't fancy going for a bike ride, I'll take a look at it.  I'll definitely aim to get something done for the next release, when there's another improvement in the 2D drawings coming in any case.
Best regards,
Dave

**Comment by greglandrum** [comment]
Should be fixed in #3380
